### PR TITLE
Added callback option

### PR DESCRIPTION
**Pull request by jesobreira**
With this approach, it's possible to use a callback function for when the beep finished beeping (`<audio>`'s `onended` event).

```javascript
new Beep(22050).play(500, 1, [Beep.utils.amplify(8000)], function() {;
   	new Beep(22050).play(600, 1, [Beep.utils.amplify(8000)]);
});
```

See the modified README.md for details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,15 +3,36 @@ Make your browser beep
...
+You can also beep multiple beeps at the same time, all together:
+
+```html
+<script type="text/javascript">
+    new Beep(22050).play(1000, 1, [Beep.utils.amplify(8000)]);
+    new Beep(22050).play(500, 1, [Beep.utils.amplify(8000)]);
+</script>
+```
+
+Or even use a callback, and beep each beep on its own time:
```
Comment (luciferous) [inline README.md]: > and beep each beep on its own time

Can we make this "and sequence beeps one after another"?

File: README.md [hunk 2]
```diff
@@ -3,15 +3,36 @@ Make your browser beep
...
+
+```html
+<script type="text/javascript">
+    new Beep(22050).play(500, 1, [Beep.utils.amplify(8000)], function() {;
```
Comment (luciferous) [inline README.md]: rm trailing ;

File: beep.js [hunk 3]
```diff
@@ -153,12 +153,15 @@ Beep.prototype = {
...
         return [header, fmtChunk, dataChunk].join("");
     },
-    play: function(freq, duration, filters) {
+    play: function(freq, duration, filters, callback) {
         filters = filters || [];
         var data = btoa(this.encode(freq, duration, filters));
         var audio = document.createElement("audio");
         audio.src = "data:audio/x-wav;base64," + data;
         audio.play();
+        if(typeof callback != 'undefined') {
```
Comment (luciferous) [inline beep.js]: add a space after `if`

**Comment by luciferous** [comment]
Looks great! Can you squash the commits too?

File: README.md (hunk 1 above)
Comment (jesobreira) [inline README.md]: Yes, of course! I've wrote this on a modified README.md I'll send now, but here's a simple example:

```html
<script type="text/javascript" src="beep.js"></script>
<script type="text/javascript">
	keys = [[660, 0.1], [660, 0.1], [660, 0.1], [510, 0.1], [660, 0.1], [770, 0.1], [1, 0.1], [380, 0.1]];
	current = 0;
	function playbeep() {
		console.log(keys[current]);
		new Beep(22050).play(keys[current][0], keys[current][1], [Beep.utils.amplify(8000)], function() {
			current++;
			playbeep();
		});
	}

	playbeep();
</script>
```

**Comment by jesobreira** [comment]
Sure, just commited!

**Comment by jesobreira** [comment]
That's done, all squashed!

File: README.md [hunk 4]
```diff
@@ -3,21 +3,66 @@ Make your browser beep
...
+```javascript
+keys = [[660, 0.1], [660, 0.1], [660, 0.1], [510, 0.1], [660, 0.1], [770, 0.1], [1, 0.1], [380, 0.1]];
+current = 0;
+function playbeep() {
+	new Beep(22050).play(keys[current][0], keys[current][1], [Beep.utils.amplify(8000)], function() {
+		current++;
+		playbeep();
+	});
+}
+playbeep();
```
Comment (luciferous) [inline README.md]: Just a nit: can we pass current into playbeep? Like this:

```javascript
var keys = [...];
function playbeep(current) {
  new Beep(22050).play(.., function() { playbeep(current + 1); });
}
playBeep(0);
```

Comment (luciferous) [inline README.md]: Might as well pass all the state in.

```javascript
function play(keys) {
  var key = keys.shift();
  new Beep(22050).play(key[0], key[1], function() { playbeep(keys); });
}
play([[660, 0.1], [660, 0.1], [660, 0.1], [510, 0.1], [660, 0.1], [770, 0.1], [1, 0.1], [380, 0.1]]);
```

Comment (jesobreira) [inline README.md]: Hi, I'm actually a little busy with college, but I'll commit that change asap!

Comment (luciferous) [inline README.md]: Thanks @jesobreira!

**Comment by jesobreira** [comment]
Done, dude! The better code example was added to README.md and the commits were squashed again!

**Pull request merged**